### DISCUSSION TITLE
Synology selfhosted Bitwaren - can't login after an update??

### DISCUSSION THREAD

**Discussion by tomekfff** [dani-garcia/vaultwarden#2493] [Q&A]
Hi Dani

Today i got a message that Bitwarden running on Synology docker stopped working. Only now I realized that I can't log in, I get the message that either user name or password is incorrect. I logged in from an admin panel and it looks like all the users have been deleted. That and all the SMTP settings from what I can tell are gone. The container itself seems to be running fine. Anything you can do to help me recover my passwords?
I can only see the log from today (attached).

**Comment by BlackDex** [comment]
I'm not sure how you configured the container.
Normally there would be a volume to store persistent data which should be preserved between restarts and updates.
It could be that something changed the location of this persistent storage (Like an update of Synology).
Please check the container configuration and if the volume date is misplaced.

If you didn't created a volume, then I'm afraid that you probably have lost all this data, unless there are some backups, but i doubt that any backup would contain this non persistent stored data.

**Reply by tomekfff** [reply]
All the other containers are running fine, no update on Synology for some time now...

**Reply by BlackDex** [reply]
Well nothing changed regarding storage or whatever. So please double-check the volume configuration.

**Reply by Tieneks** [reply]
Select the container, press the _Details_ button on top of the screen and select _Volume_.

1. What does it say under _File/Folder_ and _Mount path_?
2. What does the last column say?
3. Do you see the folder mentioned under _File/Folder_ in File Station?

_I'm not sure if you're aware, but there is personal information (email) in the file you uploaded._

**Reply by tomekfff** [reply]
Thanks Tieneks, i didn't realize that. Have removed the log file from the main post. 

1. What does it say under File/Folder and Mount path? - file/ folder - docker/vaultwarden, mount path - /config
2. What does the last column say? - type - rw
3. Do you see the folder mentioned under File/Folder in File Station? - shoot, no, it's all gone ... how come? 

Same with watchtower - all data gone... All other containers are up and running.

**Reply by Tieneks** [reply]
As far as I know - but @BlackDex can correct me on that - the volume should be connected to **/data** and not **/config**.
If you don't see the folder and files in File Station it means the files only exist inside the container.
Deleting the container would mean all files - sadly - are gone.

**Comment by sipesmedia** [comment]
Same problem here. I have the same configuration (also using Watchtower for updates):
File/ folder - docker/vaultwarden, mount path - /config

The folder is empty, but looking at my backups, it's always been empty.

**Comment by sipesmedia** [comment]


**Reply by Tieneks** [reply]
As long as the container is running, the files exist. Stopping the container means the files are no longer there.
That is how Docker containers work.

**Comment by sipesmedia** [comment]


**Reply by Tieneks** [reply]
Ah, I'm sorry. I meant to say "Deleted" or "Reset".

**Comment by sipesmedia** [comment]
So Watchtower deleted the container without carrying over any of the files?
How can I prevent this in the future?
Just by changing the mount folder to /data instead of /config? I'm fairly
new to this so I'm not sure how to make the volume persistent.

**Reply by Tieneks** [reply]
Deploy as described [here](https://github.com/dani-garcia/vaultwarden#installation) and - double or triple - check the files show up in File Station.

**Comment by BlackDex** [comment]
May i ask which version was your first version of Vaultwarden you  were using?? If that was 1.24.0, then I'm afraid the initial configuration was wrong by not having a volume configured.

**Reply by sipesmedia** [reply]
It must've been. I was using Watchtower to pull the latest images available.

**Reply by tomekfff** [reply]
Probably same here

**Comment by BlackDex** [comment]
If you still have a client actie which didn't logged out yet, you should be possible to export some of the passwords.

**Reply by sipesmedia** [reply]
No, it logged me out of all instances when the new image as installed.

**Comment by BlackDex** [comment]
FYI I have create a PR which tries to prevent these kind of issues. While we never can fully prevent this, we can at least try.
See: #2501
